tests/components/asuswrt/test_device_tracker.py

```diff
@@ -24,6 +24,18 @@
         assert not result
 
 
+async def test_network_unreachable(hass):
+    """Test creating an AsusWRT scanner without a pass or pubkey."""
+    with patch("homeassistant.components.asuswrt.AsusWrt") as AsusWrt:
+        AsusWrt().connection.async_connect = mock_coro_func(exception=OSError)
+        AsusWrt().is_connected = False
+        result = await async_setup_component(
+            hass, DOMAIN, {DOMAIN: {CONF_HOST: "fake_host", CONF_USERNAME: "fake_user"}}
+        )
+        assert result
+        assert hass.data.get(DATA_ASUSWRT, None) is None
```

Comment:
It would be better if we tested the existence of states in the state machine. When there's an error there shouldn't be any states. After the error clears, the setup should complete and states be created. That way we don't rely on the details of our implementation in the tests.

Reply:
Ok, I will keeep in consideration for next PR. Thx